### PR TITLE
chore(http-signature-utils): match JWK to Open Payments definition

### LINKED ISSUE #789: Flatten `jwk` structure for paymentPointerKeys + update `createPaymentPointerKey` mutation
https://github.com/interledger/rafiki/pull/681#discussion_r1006835706

https://github.com/interledger/rafiki/pull/786#discussion_r1030893460

Open Payments validation fails if an incorrect `jwk` is created via the `createPaymentPointerKey` admin API mutation, and then is fetched via `{{paymentPointer}}/jwks.json`. We should enforce this contract during creation.

### LINKED ISSUE #845: Don't return `keyId` from `generateTestKeys`
The returned `publicKey` includes `kid`

https://github.com/interledger/rafiki/blob/d73f7d65b245815df46cc4cfd06409c66ead7081/packages/auth/src/tests/signature.ts#L32-L36

- https://github.com/interledger/rafiki/pull/797#discussion_r1047199816

(Wait for :point_up_2: to be merged)

### PR DESCRIPTION
**Pull request by wilsonianb**
## Changes proposed in this pull request

- Remove ~SBF~ `nbf`, etc from `JWK`
- Return `JWK` from `generateTestKeys`
- Flatten jwk in `paymentPointerKeys` table 
- Enforce payment pointer `jwk` type in admin api

## Context

- fixes #845 
- fixes #789 
- partially addresses #656

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

**Comment by sabineschaller** [comment]
Are we sure we want that? I thought the package should move towards being more open.

**Comment by wilsonianb** [comment]
I don't think it makes the package less open.
This removes optional fields that don't appear to exist in [JWK specs](https://www.rfc-editor.org/rfc/rfc7517).

File: packages/backend/src/graphql/schema.graphql [hunk 1]
```diff
@@ -517,10 +525,31 @@ input CreatePaymentPointerInput {
...
+enum Crv {
+  Ed25519
+}
+
+type Jwk {
+  kid: String!
+  x: String!
+  alg: Alg!
+  kty: Kty!
+  crv: Crv!
```
[the remark this answers is not in the document]
Comment (sabineschaller) [reply packages/backend/src/graphql/schema.graphql:545]: I don't know of any.

File: packages/http-signature-utils/src/utils/jwk.ts [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
-export interface JWK extends JoseWk {
+export interface JWK {
   kid: string
-  alg: string
-  kty: string
-  crv: string
+  alg: 'EdDSA'
+  use?: 'sig'
+  kty: 'OKP'
+  crv: 'Ed25519'
```
Comment (sabineschaller) [inline packages/http-signature-utils/src/utils/jwk.ts:8]: This is what I meant when I said I thought we wanted to be less prescriptive within this package. But we can change that later. It is more consistent this way.

**Review by sabineschaller — APPROVED** [review]
Good idea moving `/paymentPointerKey` to `/open_payments/payment_pointer/key`

File: packages/http-signature-utils/src/utils/jwk.ts (hunk 2 above)
Comment (wilsonianb) [reply packages/http-signature-utils/src/utils/jwk.ts:8]: Ah, got it.
fwiw we were already enforcing ed25519 on key parsing
https://github.com/interledger/rafiki/blob/a69b4d49f57b4a2c8763fc65b7095f696f2ed201/packages/http-signature-utils/src/utils/key.ts#L12-L17

**Pull request merged**